### PR TITLE
Generalized the setup instructions to be OS agnostic

### PR DESCRIPTION
**Pull request by cngo-github**
Generalized overall setup instructions and made them OS agnostic. Added OS dependent instructions for installing JDK 8, SBT, Docker, and Kubernetes. This is a documentation change.

### CODE REVIEW

File: SETUP.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ This guide is to get you setup with the tools you'll need for development.
    3. Verify the installation: `sdk version`
 2. Install _Coretto Java 8_, [link](https://sdkman.io/usage).
    1. List the versions: `sdk list java`
-   2. Install the desired versions: `sdk install java 8.332.08.1-amzn`
+   2. Install the desired versions: `sdk install java 8.332.08.1-amzn` NOTE: This version is not available on a M1 Macbook. In that case, use: `sdk install java 8.0.333.fx-librca`
```
Comment (marcinwadon) [inline SETUP.md:15]: Is this related to Fedora running on Apple M1?

Comment (cngo-github) [reply SETUP.md:15]: It is not. It is for M1 Macbook, OS 12.4.

Comment (marcinwadon) [reply SETUP.md:15]: But this README is for:
> **Platform**: Fedora 35, 64-bit

**Review by marcinwadon — CHANGES_REQUESTED** [review]

File: SETUP.md (hunk 1 above)
Comment (marcinwadon) [reply SETUP.md:15]: also, `sdk` is not a default package manager on OS X. why not use `brew`?

**Comment by cngo-github** [comment]
I generalized the setup instructions so that they are OS agnostic.

File: SETUP.md [hunk 2]
```diff
@@ -2,42 +2,14 @@
 
 This guide is to get you setup with the tools you'll need for development.
 
-**Date**: May 24, 2002
-
-**Platform**: Fedora 35, 64-bit
```
Comment (ryle-goehausen) [inline SETUP.md:7]: I'd recommend just leaving the install instructions here and making a separate one for mac. Generally there's a setup guide per major distribution in most repos, and there's no reason to remove a previously tested workflow here.

File: SETUP.md [hunk 3]
```diff
@@ -2,42 +2,14 @@
...
-   1. `curl -s "https://get.sdkman.io" | bash`
-   2. `source "$HOME/.sdkman/bin/sdkman-init.sh"`
-   3. Verify the installation: `sdk version`
-2. Install _Coretto Java 8_, [link](https://sdkman.io/usage).
-   1. List the versions: `sdk list java`
-   2. Install the desired versions: `sdk install java 8.332.08.1-amzn`
-   3. If _JAVA_HOME_ is not set, re-source SDKMan's init script: `source "$HOME/.sdkman/bin/sdkman-init.sh"`
-3. Install the latest version of _Scala_: `sdk install scala`
-4. Install the latest version of _SBT_: `sdk install sbt`
+1. Install _Java 8_.
```
Comment (ryle-goehausen) [inline SETUP.md:5]: This should specify the version and how to get it. On mac generally brew is recommended for most installs but sort of up to the user. I would recommend specifying OpenJDK (because Oracle can have issues with the crypto libs)

File: SETUP.md [hunk 4]
```diff
@@ -2,42 +2,14 @@
...
+2. Install _Scala_.
```
Comment (ryle-goehausen) [inline SETUP.md:6]: SBT will download scala for you automatically, not required

File: SETUP.md [hunk 5]
```diff
@@ -2,42 +2,14 @@
...
+3. Install _SBT_.
```
Comment (ryle-goehausen) [inline SETUP.md:7]: Specify link to how to install (on mac)

File: SETUP.md (hunk 3 above)
Comment (ryle-goehausen) [reply SETUP.md:5]: Should include a link to a guide here.

File: SETUP.md (hunk 2 above)
Comment (cngo-github) [reply SETUP.md:7]: @marcinwadon wants to avoid that, preferring to generalize the instructions.

File: SETUP.md (hunk 4 above)
Comment (cngo-github) [reply SETUP.md:6]: Removed the instruction to install Scala.

File: SETUP.md (hunk 3 above)
Comment (cngo-github) [reply SETUP.md]: @marcinwadon wants to avoid OS specific instructions. I did add a link to OpenJdk 8 though

File: SETUP.md (hunk 5 above)
Comment (cngo-github) [reply SETUP.md]: @marcinwadon would like to keep OS specific instructions out of this guide. I added a link to Scala-SBT though.

File: SETUP.md [hunk 6]
```diff
@@ -2,42 +2,14 @@
...
-5. Login to _Docker_, [link](https://docs.docker.com/desktop/linux/#credentials-management).
-6. Install _docker-credential-pass_, [link](https://github.com/docker/docker-credential-helpers/releases).
-   1. Download the appropriate tarball for your machine and extract it.
-   2. `sudo cp /path/to/docker-credential-pass /bin/docker-credential-desktop`
-7. Install _kubectl_, [link](https://kubernetes.io/docs/tasks/tools/install-kubectl-linux/).
-   1. Ensure that _kubectl_ is pointing to the Docker Desktop context:
-      1. `kubectl config get-contexts`
-      2. `kubectl config use-context docker-desktop`
+1. Install _Docker Desktop_, [link](https://docs.docker.com/desktop/).
+2. Enable _Kubernetes_ on _Docker_.
```
[the remark this answers is not in the document]
Comment (cngo-github) [reply SETUP.md]: Added a link to Docker's docs.

File: SETUP.md (hunk 2 above)
Comment (cngo-github) [reply SETUP.md:7]: Re-added OS specific links and instructions in separate sections.

**Review by tbekas — CHANGES_REQUESTED** [review]

File: SETUP.md [hunk 7]
```diff
@@ -1,43 +1,57 @@
...
+
+Installing _Java 8_ can vary depending on your operating system and chosen methodology. The recommended version to install is _OpenJDK 8_.
+
+## Fedora Linux
+
+Installing via _DNF_ is the recommended method for Fedora Linux. Instructions for that can be found [here](https://docs.fedoraproject.org/en-US/quick-docs/installing-java/).
+
+## Mac OS
+
+Installing via [_Homebrew_](https://brew.sh/) is the recommended method for Mac, though it can be used for other operating systems as well. To install _OpenJDK 8_ on a machine with an Intel chip, go [here](https://formulae.brew.sh/formula/openjdk@8). _OpenJDK 8_ is not available for the M1 chipset. So in that case, _OpenJDK 11_ can be used, [link](https://formulae.brew.sh/formula/openjdk@11#default).
```
Comment (marcinwadon) [inline SETUP.md:39]: > _OpenJDK 8_ is not available for the M1 chipset.

OpenJDK 8 is preferable in any case. On M1 chipset one can install AdoptOpenJDK or Azul Zulu build.

image

Comment (marcinwadon) [reply SETUP.md:39]: As discussed internally, we would like to bump docs to Java 11 :+1:

File: SETUP.md [hunk 8]
```diff
@@ -1,43 +1,57 @@
...
+3. Install [_kubectl_](https://docs.docker.com/desktop/kubernetes/).
+
+Alternatively, simply install [_Docker Desktop_](https://docs.docker.com/desktop/linux/install/fedora/) and enable [_Kubernetes_](https://docs.docker.com/desktop/kubernetes/).
+
+## MacOS
+
+1. Install [_Docker Desktop_](https://docs.docker.com/desktop/mac/install/).
+2. Enable _Kubernetes_ via _Docker Desktop_, [link](https://docs.docker.com/desktop/kubernetes/).
+
+# Installing Java 8
```
[the remark this answers is not in the document]
Comment (cngo-github) [reply SETUP.md]: Updated the docs with install instructions for Java 11.

File: SETUP.md (hunk 7 above)
Comment (cngo-github) [reply SETUP.md]: Updated the docs with install instructions for Java 11.

**Review by marcinwadon — APPROVED** [review]

**Review by tbekas — APPROVED** [review]
